tripal/src/Plugin/TripalPubLibrary/TripalPubLibraryPubMed.php

```diff
@@ -0,0 +1,1249 @@
+              $pub['Publication Date'] = $medline;
+            }
+            else {
+              $pub['Publication Date'] = "Date Unknown";
+            }
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  }
+  
+  /**
+   * Parses the section from the XML returned from PubMed that contains
+   * information regarding to dates
+   *
+   * @param $xml
+   *   The XML to parse
+   * @param $pub
+   *   The publication object to which additional details will be added
+   *
+   * @ingroup tripal_pub
+   */
+  private function pmidParseDate($xml, $element_name) {
+    $date = [];
+  
+    while ($xml->read()) {
+      $element = $xml->name;
+  
+      if ($xml->nodeType == \XMLReader::END_ELEMENT and $element == $element_name) {
+        // if we're at the </$element_name> then we're done
+        return $date;
+      }
+      if ($xml->nodeType == \XMLReader::ELEMENT) {
+        switch ($element) {
+          case 'Year':
+            $xml->read();
+            $date['year'] = $xml->value;
+            break;
+          case 'Month':
+            $xml->read();
+            $month =
+            $date['month'] = $xml->value;
+            break;
+          case 'Day':
+            $xml->read();
+            $date['day'] = $xml->value;
+            break;
+          case 'MedlineDate':
+            // the medline date is when the date cannot be broken into distinct month day year.
+            $xml->read();
+            if (!$date['year']) {
+              $date['year'] = preg_replace('/^.*(\d{4}).*$/', '\1', $xml->value);
+            }
+            $date['medline'] = $xml->value;
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  }
+  
+  /**
+   * Parses the section from the XML returned from PubMed that contains
+   * information about the author list for a publication
+   *
+   * @param $xml
+   *   The XML to parse
+   * @param $pub
+   *   The publication object to which additional details will be added
+   *
+   * @ingroup tripal_pub
+   */
+  private function pmidParseAuthorlist($xml, &$pub) {
+    $num_authors = 0;
+  
+    while ($xml->read()) {
+      $element = $xml->name;
+  
+      if ($xml->nodeType == \XMLReader::END_ELEMENT) {
+        // if we're at the </AuthorList> element then we're done with the article...
+        if ($element == 'AuthorList') {
+          // build the author list before returning
+          $authors = '';
+          foreach ($pub['Author List'] as $author) {
+            if ($author['valid'] == 'N') {
+              // skip non-valid entries.  A non-valid entry should have
+              // a corresponding corrected entry so we can saftely skip it.
+              continue;
+            }
+            if (array_key_exists('Collective', $author)) {
+              $authors .= $author['Collective'] . ', ';
+            }
+            else {
+              $authors .= $author['Surname'] . ' ' . $author['First Initials'] . ', ';
```

Comment:
```suggestion
              $authors .= ($author['Surname']??'') . ' ' . ($author['First Initials']??'') . ', ';
```
To see why I am adding this, search in "Abstract" with term "carrot" - without this change you will see the error:

Warning: Undefined array key "First Initials" in Drupal\tripal\Plugin\TripalPubLibrary\TripalPubLibraryPubmed->pmidParseAuthorlist() (line 1119 of modules/contrib/tripal/tripal/src/Plugin/TripalPubLibrary/TripalPubLibraryPubMed.php).

Drupal\tripal\Plugin\TripalPubLibrary\TripalPubLibraryPubmed->pmidParseAuthorlist(Object, Array) (Line: 751)